### DISCUSSION TITLE
New shape instance render latency

### DISCUSSION THREAD

**Discussion by kazcw** [enso-org/enso#6716] [Help and troubleshooting]
When I create a new instance of a shape, I'm observing a 1-frame delay before it starts drawing. This is not caused by GPU buffer upload; other GPU parameter changes are visible without the delay. I suspect it is related to asynchronous processing of changes in the display object hierarchy. Is this expected behavior? It is causing a flicker in the new edge implementation when I need to change shaders in response to hover position changes (#6707), and working around it by delaying hiding the previous shader would add significant complexity.

**Comment by wdanilo** [comment]
@kazcw I have observed the same thing recently in one of my debug scenes. I was not digging deeply into it then, but yes, this is a bug and should be fixed. TBH I don't know why it's happening, but I'm almost sure it was not happening in the past (few months ago). Do you think you can track it down? If it blocks you from finishing edges properly, then maybe its time to kill it :)

**Comment by kazcw** [comment]
Ok, I'll look in to it. Issue: #6717
